base/shell/explorer/syspager.cpp

```diff
@@ -617,7 +619,8 @@ void CBalloonQueue::Show(Info& info)
 
     // TODO: NIF_REALTIME, NIIF_NOSOUND, other Vista+ flags
 
-    const int index = IndexOf(info.pSource);
+    m_current = info.pSource;
+    const int index = IndexOf(m_current);
```

Comment:
note: I even wonder whether it could be possible to call the IndexOf(...) directly as the 1st parameter of the GetItemRect call below (optional).

Reply:
Done.